### PR TITLE
chore: v2.3.0 release update

### PR DESCRIPTION
**Pull request by fernando-cortez**
### Description
Updating changelog and project version for v2.3.0 release.

### Issue Number(s)
N/A

### Contribution checklist
 - [ N/A ] Tests have been added for boss room and/or utilities pack
 - [x] Release notes have been added to the [project changelog](../CHANGELOG.md) file and/or [package changelog](../Packages/com.unity.multiplayer.samples.coop/CHANGELOG.md) file
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [ N/A ] JIRA ticket ID is in the PR title or at least one commit message
 - [ N/A ] Include the ticket ID number within the body message of the PR to create a hyperlink
 - [ N/A ] An Index entry has been added in readme.md if applicable

### CODE REVIEW

File: Packages/com.unity.multiplayer.samples.coop/package.json [hunk 1]
```diff
@@ -7,9 +7,9 @@
   "unity": "2020.3",
   "description": "Utilities package built on top of Netcode for GameObjects, providing useful scripts and tools.",
   "dependencies": {
-    "com.unity.learn.iet-framework": "1.2.1",
-    "com.unity.multiplayer.tools": "1.1.0",
-    "com.unity.netcode.gameobjects": "1.4.0",
-    "com.unity.services.relay": "1.0.3"
+    "com.unity.learn.iet-framework": "3.1.3",
+    "com.unity.multiplayer.tools": "2.0.0-pre.3",
```
Comment (LPLafontaineB) [inline Packages/com.unity.multiplayer.samples.coop/package.json:11]: is the pre-release version required for this package, or can it stay with 1.1.0?

Comment (fernando-cortez) [reply Packages/com.unity.multiplayer.samples.coop/package.json:11]: It's not required, no. Are there risks to upgrading Tools-side?

Comment (LPLafontaineB) [reply Packages/com.unity.multiplayer.samples.coop/package.json:11]: I don't think so, apart from it being in pre-release

Comment (fernando-cortez) [reply Packages/com.unity.multiplayer.samples.coop/package.json]: For the sake of getting this PR in, it's been reverted. I've also asked Product to take a look!

**Review by LPLafontaineB — APPROVED** [review]

**Pull request merged**